x.payment_batching/payment_batching.md

```diff
@@ -0,0 +1,307 @@
+implementing changeless transactions may also be small.  Your goal is to
+find a single input you own whose value is equal to or slightly larger
+than the total value of the outputs you want to send.  A simple
+algorithm would be to wait until you have a minimum number of payments
+to batch and check to see if you have an input that is equal to or
+slightly larger than that.  If not, wait until you have another payment
+to send, combine that with the earlier payments, and try again.  Keep
+trying until you reach a limit (e.g. it's time to send the oldest
+payment) or the aggregate output value is larger than your largest
+input.
+
+For more information about changeless transactions, including exactly
+what "slightly larger than the output value" means, see the
+corresponding section in the chapter about [coin selection
+strategies][].
+
+### Replace-by-Fee (RBF)
+
+RBF is typically promoted as a strategy for increasing the fees paid in
+a transaction so that it confirms sooner, but it also allows for adding
+outputs to a transaction, making it suitable for adding even more
+payments to a previously-sent transaction.
+
+However, RBF transactions must pay a higher feerate than the original
+transaction.  This means replacing a large batch payment costs more than
+replacing a small independent payment.  Since fee batching benefits more
+from larger transactions, the cost increases of RBF act in direct
+opposition to the fee savings of payment batching.
+
+Looking at the savings rate charts above, we can consider some examples.
+If you sent a single payment and want to turn that into a batch
+containing two payments, you'll expect to save 40% from batching in the
+best case---so paying a fee increase of anything up to 40% can save you
+money overall.  Alternatively, if you already have a batch of ten
+payments and are considering adding an eleventh, the extra savings is
+only about 1%, so it might not be worth it if RBF requires you increase
+the overall feerate of the transaction by more than 1%.
+
+However, there is a time when it's guaranteed to be more efficient to
+add new payments to an existing transaction: when you plan to fee bump
+that existing transaction anyway.  If you sent a payment some time ago
+and are planning to use RBF to increase its fee rate, you should also
+add any additional queued payments to the replacement transaction so
+that they gain from batched efficiency.
+
+For more information about RBF, see the chapter about [fee bumping][].
+
+## Recommendations summary
+
+1. Try to create systems where your users and customers don't expect
+   their payments immediately but are willing to waiting for some time
+   (the longer the better).
+
+2. Use low-feerate consolidations to keep some large inputs available
+   for spending.
+
+3. Within each time window, send all payments together in the same
+   transaction.  Ideally, your prior consolidations should allow the
+   transaction to contain only a single input.
```

Comment:
Again, I don't understand the point about 'single input' here. If you're paying during a period of relatively high fees, then you want to reduce the number of inputs, but in general I don't think it matters when you use up your UTXOs.

Reply:
I don't understand your criticism here.  Or, rather, I *think* I understand it, but it seems to be based on not understanding how consolidations reduce fees---something I know you understand and something that this article illustrates in conjunction with batching.  So now I'm sure *I'm* missing something.  Help please!

Reply:
I think @jnewbery is just saying that batching is still generally a big win even if multiple inputs are necessary to create the transaction, and even if UTXO consolidation is generally being performed.

Reply:
ah. Sorry, I was confusing this comment about a single input with the comment in the _change avoidance_ section about a single inputs (https://github.com/bitcoinops/scaling-book/pull/18#discussion_r260981366).

I still think this comment about consolidation isn't necessary. I think UTXO consolidation is orthogonal to the benefits of batching, or am I missing something?

Reply:
Had a good out-of-band discussion with @jnewbery about this and we decided to (1) leave this unchanged for now, (2) I'll write a chapter about consolidation next, and (3) we'll re-evaluate the consolidation-related parts of this chapter based on what we put in the consolidation chapter.